tools/nut-scanner/scan_xml_http.c

```diff
@@ -221,14 +257,22 @@ nutscan_device_t * nutscan_scan_xml_http(long usec_timeout)
 				(*nut_ne_xml_push_handler)(parser, startelm_cb,
 							NULL, NULL, nut_dev);
 				(*nut_ne_xml_parse)(parser, buf, recv_size);
+				int parserFailed = (*nut_ne_xml_failed)(parser); // 0 = ok, nonzero = fail
 				(*nut_ne_xml_destroy)(parser);
 
-				nut_dev->driver = strdup("netxml-ups");
-				sprintf(buf,"http://%s",string);
-				nut_dev->port = strdup(buf);
+				if (parserFailed == 0) {
+					nut_dev->driver = strdup("netxml-ups");
+					sprintf(buf,"http://%s",string);
+					nut_dev->port = strdup(buf);
 
-				current_nut_dev = nutscan_add_device_to_device(
+					current_nut_dev = nutscan_add_device_to_device(
 						current_nut_dev,nut_dev);
+				}
+				else
+				{
+					fprintf(stderr,"Device replied with NetXML but was not deemed compatible\n");
+					return NULL;
```

Comment:
Should be fixed in https://github.com/networkupstools/nut/pull/324/commits/0535c01ef55eca3d35c16b5dab2c1baf3eb4b25f and improved in https://github.com/networkupstools/nut/pull/324/commits/1ea2f99a226629a21f55802e1333ab02e7a34e5b